rs/nns/governance/src/governance/tla/mod.rs

```diff
@@ -245,42 +290,39 @@ pub fn check_traces() {
         panic!("bad apalache bin from 'TLA_APALACHE_BIN': '{:?}'", apalache);
     }
 
-    print_stats(&traces);
-
-    let chunk_size = 20;
-    let all_pairs = traces.into_iter().flat_map(|t| {
-        t.state_pairs
-            .into_iter()
-            .filter(is_under_limit)
-            .map(move |p| (t.update.clone(), t.constants.clone(), p))
-    });
-    let chunks = all_pairs.chunks(chunk_size);
-    for chunk in &chunks {
-        let mut handles = vec![];
-        for (update, constants, pair) in chunk {
-            let apalache = apalache.clone();
-            let constants = constants.clone();
-            let pair = pair.clone();
-            // NOTE: We adopt the convention to reuse the 'process_id" as the tla module name
-            let tla_module = format!("{}_Apalache.tla", update.process_id);
-            let tla_module = get_tla_module_path(&tla_module);
-
-            let handle = thread::spawn(move || {
-                check_tla_code_link(
-                    &apalache,
-                    PredicateDescription {
-                        tla_module,
-                        transition_predicate: "Next".to_string(),
-                        predicate_parameters: Vec::new(),
-                    },
-                    pair,
-                    constants,
-                )
-            });
-            handles.push(handle);
+    // A poor man's parallel_map; process up to MAX_THREADS state pairs in parallel. Use mpsc channels
```

Comment:
Would [threadpool] work for you?

[threadpool]: https://crates.io/crates/threadpool